### ISSUE TITLE
Webpack fails on CE 4.0.65

### ISSUE THREAD

**Issue by ghoussard** [akeneo/pim-community-dev#13061]
:bug: **I'm reporting a Bug** :bug:

Hello the team,

I'm using Akeneo CE 4.0.65.

My CLI prints me many errors when I run webpack or webpack-dev (through docker) : 

```sh
docker-compose run -u node --rm node yarn run webpack-dev
Creating pim_node_run ... done
yarn run v1.22.4
$ yarn requirements && NODE_PATH=node_modules webpack --config $npm_package_config_source/webpack.config.js
$ node $npm_package_config_source/frontend/build/check-requirements.js
Checking PIM frontend requirements
Starting webpack from /srv/pim in dev mode
Executing pre-build scripts
(node:47) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
$ node $npm_package_config_styles
$ node $npm_package_config_source/frontend/build/update-extensions.js

Starting LESS compilation

‣ vendor/akeneo/pim-community-dev/src/Oro/Bundle/PimDataGridBundle/Resources/public/less/index.less
‣ vendor/akeneo/pim-community-dev/src/Akeneo/Platform/Bundle/UIBundle/Resources/public/less/index.less
‣ vendor/akeneo/pim-community-dev/src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/public/less/index.less


Updating form extensions.json
✓ Saved CSS to public/css/pim.css
Version: webpack 4.29.6
Time: 38666ms
Built at: 10/21/2020 9:06:08 AM
                                      Asset       Size  Chunks             Chunk Names
assets/41db83c1c1d72900f79ef0cf71bb0937.svg  573 bytes          [emitted]  
assets/669652e3db73e445544c39e1d428e100.svg   1.48 KiB          [emitted]  
assets/75daeed2d5f5d8530728812ebaa79be1.svg   3.28 KiB          [emitted]  
assets/93b575e6d38167ac3865176e1e7d2603.svg   63.8 KiB          [emitted]  
assets/999aefd96c0c9c7b25d9c72feaf1c600.svg     94 KiB          [emitted]  
                                main.min.js   3.44 MiB    main  [emitted]  main
                            main.min.js.map    3.5 MiB    main  [emitted]  main
                              vendor.min.js   4.16 MiB  vendor  [emitted]  vendor
                          vendor.min.js.map    4.2 MiB  vendor  [emitted]  vendor
Entrypoint main = vendor.min.js vendor.min.js.map main.min.js main.min.js.map

ERROR in /srv/pim/public/bundles/akeneoconnectivityconnection-react/settings/components/credentials/Credential.tsx
./public/bundles/akeneoconnectivityconnection-react/settings/components/credentials/Credential.tsx
[tsl] ERROR in /srv/pim/public/bundles/akeneoconnectivityconnection-react/settings/components/credentials/Credential.tsx(37,15)
      TS2615: Type of property 'defaultProps' circularly references itself in mapped type 'Pick<ForwardRefExoticComponent<Pick<Pick<any, Exclude<keyof ReactDefaultizedProps<StyledComponentInnerComponent<WithC>, ComponentPropsWithRef<StyledComponentInnerComponent<WithC>>>, StyledComponentInnerAttrs<...> | ... 1 more ... | StyledComponentInnerAttrs<...>> | Exclude<...> | Exclude<...> | Exclude<...>> & Parti...'.

ERROR in /srv/pim/public/bundles/akeneoconnectivityconnection-react/settings/components/credentials/Credential.tsx
./public/bundles/akeneoconnectivityconnection-react/settings/components/credentials/Credential.tsx
[tsl] ERROR in /srv/pim/public/bundles/akeneoconnectivityconnection-react/settings/components/credentials/Credential.tsx(37,15)
      TS2615: Type of property 'defaultProps' circularly references itself in mapped type 'Pick<ForwardRefExoticComponent<Pick<Pick<any, Exclude<keyof ReactDefaultizedProps<StyledComponentInnerComponent<WithC>, ComponentPropsWithRef<StyledComponentInnerComponent<WithC>>>, any>> & Partial<...>, any> & { ...; } & WithChildrenIfReactComponentClass<...>>, "$$typeof" | ... 2 more ... | "propTypes">'.

ERROR in /srv/pim/public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts
./public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts
[tsl] ERROR in /srv/pim/public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts(27,29)
      TS2783: 'clientId' is specified more than once, so this usage will be overwritten.

ERROR in /srv/pim/public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts
./public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts
[tsl] ERROR in /srv/pim/public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts(27,29)
      TS2783: 'clientId' is specified more than once, so this usage will be overwritten.

ERROR in /srv/pim/public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts
./public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts
[tsl] ERROR in /srv/pim/public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts(28,29)
      TS2783: 'secret' is specified more than once, so this usage will be overwritten.

ERROR in /srv/pim/public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts
./public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts
[tsl] ERROR in /srv/pim/public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts(28,29)
      TS2783: 'secret' is specified more than once, so this usage will be overwritten.

ERROR in /srv/pim/public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts
./public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts
[tsl] ERROR in /srv/pim/public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts(29,29)
      TS2783: 'username' is specified more than once, so this usage will be overwritten.

ERROR in /srv/pim/public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts
./public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts
[tsl] ERROR in /srv/pim/public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts(29,29)
      TS2783: 'username' is specified more than once, so this usage will be overwritten.

ERROR in /srv/pim/public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts
./public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts
[tsl] ERROR in /srv/pim/public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts(31,29)
      TS2783: 'userRoleId' is specified more than once, so this usage will be overwritten.

ERROR in /srv/pim/public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts
./public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts
[tsl] ERROR in /srv/pim/public/bundles/akeneoconnectivityconnection-react/settings/reducers/connections-reducer.ts(31,29)
      TS2783: 'userRoleId' is specified more than once, so this usage will be overwritten.

ERROR in /srv/pim/public/bundles/pimdatagrid/js/datagrid/column-selector.ts
./public/bundles/pimdatagrid/js/datagrid/column-selector.ts
[tsl] ERROR in /srv/pim/public/bundles/pimdatagrid/js/datagrid/column-selector.ts(466,5)
      TS2322: Type 'Partial<{ [name: string]: Column; }>' is not assignable to type '{ [name: string]: Column; }'.
  Index signatures are incompatible.
    Type 'Column | undefined' is not assignable to type 'Column'.
      Type 'undefined' is not assignable to type 'Column'.

ERROR in /srv/pim/public/bundles/pimui/js/view/base.ts
./public/bundles/pimui/js/view/base.ts
[tsl] ERROR in /srv/pim/public/bundles/pimui/js/view/base.ts(94,28)
      TS2538: Type 'undefined' cannot be used as an index type.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
make: *** [Makefile:41 : javascript-dev] Erreur 2
```

Node and PHP images are up to date.

This issue is blocking because it causes failure of my docker prod image build.

Thanks for you help.
[akeneo-pim-system-info_2020-10-21_10_14.txt](https://github.com/akeneo/pim-community-dev/files/5415121/akeneo-pim-system-info_2020-10-21_10_14.txt)

**Comment by ghoussard** [comment]
I solved two last errors by installing @types/underscore (https://github.com/akeneo/pim-community-dev/issues/12698#issuecomment-694799071) :

```sh
ERROR in /srv/pim/public/bundles/pimdatagrid/js/datagrid/column-selector.ts
./public/bundles/pimdatagrid/js/datagrid/column-selector.ts
[tsl] ERROR in /srv/pim/public/bundles/pimdatagrid/js/datagrid/column-selector.ts(466,5)
      TS2322: Type 'Partial<{ [name: string]: Column; }>' is not assignable to type '{ [name: string]: Column; }'.
  Index signatures are incompatible.
    Type 'Column | undefined' is not assignable to type 'Column'.
      Type 'undefined' is not assignable to type 'Column'.

ERROR in /srv/pim/public/bundles/pimui/js/view/base.ts
./public/bundles/pimui/js/view/base.ts
[tsl] ERROR in /srv/pim/public/bundles/pimui/js/view/base.ts(94,28)
      TS2538: Type 'undefined' cannot be used as an index type.
```

**Comment by timonburkard** [comment]
Hi @ghoussard,

I had the same problem some days ago.

Try to executed the following commands:
```
bin/console cache:clear --env=prod --no-warmup
bin/console pim:installer:assets --symlink --clean
yarn install
yarn run webpack
```

`yarn install` is required in case new js dependencies are used in the UI.

**Comment by ghoussard** [comment]
This issue seems to be fixed with version 4.0.66 or 4.0.67. I don't have trouble this morning after composer upgrade. Thank for the fix!
